src/python/pants/engine/process.py

```diff
@@ -67,6 +93,7 @@ class Process:
     jdk_home: str | None
     execution_slot_variable: str | None
     concurrency_available: int
+    concurrency: ProcessConcurrency | None
```

Comment:
`concurrency_available` should be deprecated in favor of `concurrency`, what is the best way for me to do that?

Reply:
To clarify, that check can be in a followup PR (and deprecation in a future followup after that). It doesn't have to be in this one.